server/src/main/java/org/apache/iotdb/db/engine/querycontext/ReadOnlyMemChunk.java

```diff
@@ -18,106 +18,69 @@
  */
 package org.apache.iotdb.db.engine.querycontext;
 
-import java.util.Collections;
-import java.util.Iterator;
+import java.io.IOException;
 import java.util.List;
 import java.util.Map;
-import org.apache.iotdb.db.engine.memtable.MemSeriesLazyMerger;
-import org.apache.iotdb.db.engine.memtable.TimeValuePairSorter;
 import org.apache.iotdb.db.query.reader.MemChunkLoader;
-import org.apache.iotdb.db.utils.MathUtils;
-import org.apache.iotdb.tsfile.read.TimeValuePair;
+import org.apache.iotdb.db.query.reader.universal.PriorityMergeReader;
+import org.apache.iotdb.db.utils.datastructure.TVList;
 import org.apache.iotdb.tsfile.common.conf.TSFileDescriptor;
 import org.apache.iotdb.tsfile.encoding.encoder.Encoder;
 import org.apache.iotdb.tsfile.file.metadata.ChunkMetaData;
 import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
 import org.apache.iotdb.tsfile.file.metadata.statistics.Statistics;
-import org.apache.iotdb.tsfile.utils.TsPrimitiveType.TsDouble;
-import org.apache.iotdb.tsfile.utils.TsPrimitiveType.TsFloat;
+import org.apache.iotdb.tsfile.read.IPointReader;
+import org.apache.iotdb.tsfile.read.TimeValuePair;
 
 //TODO: merge ReadOnlyMemChunk and WritableMemChunk and IWritableMemChunk
-public class ReadOnlyMemChunk implements TimeValuePairSorter {
+public class ReadOnlyMemChunk {
 
-  private boolean initialized;
   private String measurementUid;
   private TSDataType dataType;
-  private TimeValuePairSorter memSeries;
-  private List<TimeValuePair> sortedTimeValuePairList;
 
   Map<String, String> props;
   private int floatPrecision = TSFileDescriptor.getInstance().getConfig().getFloatPrecision();
+  private ChunkMetaData cachedMetaData;
+
+  private PriorityMergeReader mergeReader;
+  private PriorityMergeReader chunkedReader;
```

Comment:
I suggest using one ReadOnlyMemChunk to store one chunk in a memtable. The TsFileProcessor return a Pair<List<ReadOnlyMemtable>, List<ChunkMetadata>>
Init the chunk metadata in the constructor in this class and cache it. 
Therefore, we only need one chunkreader, and it should not be a mergereader, we could directly get the  IPointReader in TVList.
In the AbstractMemTable, the version should be init by Long.MAX_VALUE for a working memtable, the memtable version is set only when it comes to a  flushing memtable.

Reply:
this is a pretty nice design